Indexing/IndexManager.cpp

```diff
@@ -91,9 +56,6 @@ void IndexManager::release(IndexType t)
 
   e.refCnt--;
   if(e.refCnt==0) {
-    if(t==GENERATING_SUBST_TREE) {
```

Comment:
Can we take this opportunity to chaneg the name of the index? Perhaps to `BinaryResolutionTree`?

Reply:
A good idea! I will add some name changes to the PR...